### PR TITLE
solution

### PR DESCRIPTION
**Pull request by uzlss**

### CODE REVIEW

File: app/chivalry.py [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+        self.hp: int = knight.get("hp")
+        self.armour: list[dict] = knight.get("armour")
+        self.weapon: dict = knight.get("weapon")
+        self.potion = knight.get("potion")
+        self.protection = 0
+
+    def apply_equipment(self) -> None:
+        self.protection = sum(
+            [armour.get("protection") for armour in self.armour]
+        )
```
Comment (Vasyl-Smutok) [inline app/chivalry.py:21]:
```suggestion
        self.protection = sum(armour.get("protection") for armour in self.armour)

```

File: app/chivalry.py [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+        self.power += self.weapon.get("power")
+        if self.potion:
+            for key in self.potion["effect"]:
+                self.__dict__[key] += self.potion["effect"][key]
```
Comment (Vasyl-Smutok) [inline app/chivalry.py:25]: use `setattr` and  `getattr` here

File: app/chivalry.py [hunk 3]
```diff
@@ -0,0 +1,42 @@
+from __future__ import annotations
+
+
+class Chivalry:
```
Comment (Vasyl-Smutok) [inline app/chivalry.py:4]: I think `Knight` better name

image

File: app/chivalry.py [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+
+    def duel(self, other: Chivalry) -> None:
+        self.apply_equipment()
+        other.apply_equipment()
+        self.hp = Chivalry.health_checker(
+            self.hp - (other.power - self.protection)
+        )
+        other.hp = Chivalry.health_checker(
+            other.hp - (self.power - other.protection)
+        )
```
Comment (Vasyl-Smutok) [inline app/chivalry.py:35]: add return 

```
        return {
            self.name: self.hp,
            other.name: other.hp
        }
```

File: app/main.py [hunk 5]
```diff
@@ -1,213 +1,19 @@
...
+    arthur.duel(red_knight)
     return {
-        lancelot["name"]: lancelot["hp"],
-        arthur["name"]: arthur["hp"],
-        mordred["name"]: mordred["hp"],
-        red_knight["name"]: red_knight["hp"],
+        lancelot.name: lancelot.hp,
+        arthur.name: arthur.hp,
+        mordred.name: mordred.hp,
+        red_knight.name: red_knight.hp,
```
Comment (Vasyl-Smutok) [inline app/main.py:16]: what if you have 30 duel?

it is better to move it to duel

**Review by Vasyl-Smutok — CHANGES_REQUESTED** [review]

File: app/chivalry.py (hunk 1 above)
Comment (uzlss) [reply app/chivalry.py:21]: flake8 is arguing with it :)

File: app/knight.py [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+        self.protection = 0
+
+    def apply_equipment(self) -> None:
+        self.protection = sum(
+            [
+                armour.get("protection")
+                for armour
+                in self.armour
+            ]
+        )
```
Comment (Vasyl-Smutok) [inline app/knight.py:25]: no need to create list
```suggestion
        self.protection = sum(
            armour.get("protection")
            for armour in self.armour
        )

```

**Review by Vasyl-Smutok — CHANGES_REQUESTED** [review]

**Review by Vasyl-Smutok — APPROVED** [review]
Good Job 👍

**Pull request closed without merging**